### ISSUE TITLE
AES New Encryptor doesn't handle empty string

### ISSUE THREAD

**Issue by jhjwind** [jmazzi/crypt_keeper#109]
According to the documentation [provider/aes_new.rb](https://github.com/jmazzi/crypt_keeper/blob/master/lib/crypt_keeper/provider/aes_new.rb#L21), the aes_new returns empty string when encounters empty string, instead it raises exception

```
2.3.0 :036 > @aes_new_encryptor.encrypt("")
ArgumentError: data must not be empty
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/aes-0.5.0/lib/aes/aes.rb:119:in `update'
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/aes-0.5.0/lib/aes/aes.rb:119:in `_encrypt'
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/aes-0.5.0/lib/aes/aes.rb:58:in `encrypt'
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/aes-0.5.0/lib/aes/aes.rb:5:in `encrypt'
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/crypt_keeper-0.20.0/lib/crypt_keeper/provider/aes_new.rb:27:in `encrypt'
    from (irb):36
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/railties-4.1.7/lib/rails/commands/console.rb:90:in `start'
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/railties-4.1.7/lib/rails/commands/console.rb:9:in `start'
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:69:in `console'
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:40:in `run_command!'
    from /Users/hanjie/.rvm/gems/ruby-2.3.0/gems/railties-4.1.7/lib/rails/commands.rb:17:in `<top (required)>'
    from bin/rails:4:in `require'
    from bin/rails:4:in `<main>'
```

**Comment by jesperronn** [comment]
:+1:  for the fix. It bites me currently and prevents me from upgrading Rails

**Comment by danielricecodes** [comment]
I think the fix in #110 might cause issues using `search_by_plaintext`.  If a blank string leaks into my table, I can no longer use `search_by_plaintext`.  If I delete any records with a blank string - I can scan my table again.



```bash
[6] pry(main)> Lessee.search_by_plaintext(:ssn, '123').any?
  Lessee Exists (353.1ms)  SELECT  1 AS one FROM "lessees" WHERE pgp_sym_decrypt([FILTERED]) LIMIT $1  [["LIMIT", 1]]
=> false
[7] pry(main)> ssn.update_attributes(ssn: '')
   (0.2ms)  BEGIN
  SQL (0.4ms)  UPDATE "lessees" SET "updated_at" = $1, "ssn" = $2 WHERE "lessees"."id" = $3  [["updated_at", "2018-01-17 21:07:29.278633"], ["ssn", ""], ["id", 1111]]
   (1.9ms)  COMMIT
=> true
[8] pry(main)> Lessee.search_by_plaintext(:ssn, '123').any?
  Lessee Exists (120.4ms)  SELECT  1 AS one FROM "lessees" WHERE pgp_sym_decrypt([FILTERED]) LIMIT $1  [["LIMIT", 1]]
ActiveRecord::StatementInvalid: PG::ExternalRoutineInvocationException: ERROR:  Wrong key or corrupt data
: SELECT  1 AS one FROM "lessees" WHERE ((pgp_sym_decrypt(cast("ssn" AS bytea), '[MANUALLY FILTERED') = '123')) LIMIT $1
from /Users/danielrice/.rvm/gems/ruby-2.4.2/gems/rack-mini-profiler-0.10.5/lib/patches/db/pg.rb:90:in `async_exec'
```
